src/app/hm-admin/services/api/api.service.ts

```diff
@@ -0,0 +1,63 @@
+import {Injectable} from '@angular/core';
+import {Http} from '@angular/http';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/mergeMap';
+import {ConfigService} from '../config/Config.service';
+import {SchemaService} from '../schema/schema.service';
+import {EntryPoint} from '../models/EntryPoint';
+import {entrypointHelper} from '../../helper/entrypointHelper';
+import {stringHelper} from '../../helper/stringHelper';
+import {PagedCollection} from '../models/PagedCollection';
+import {CoreDefinitionService} from '../core/CoreDefinition.service';
+import {Observable, BehaviorSubject} from 'rxjs';
+
+
+@Injectable()
+export class APIService {
+
+  public entrypoints: Array<EntryPoint> = [];
+
+  constructor(private http: Http, private config: ConfigService, private schema: SchemaService, private coreDefinitionService: CoreDefinitionService) {
+    this.schema.entrypoints.subscribe(entrypoints => this.entrypoints = entrypoints);
+  }
+
+  /**
+   * Get collection by URL
+   *
+   * @param {string} path
+   *
+   * @returns {Observable<PagedCollection|null>}
+   */
+  getCollectionByUrl(path: string, definition: any) {
+    let query = this.http.get(path)
+      .map(data => data.json())
+      .map(collection => new PagedCollection(collection, definition));
+
+    query.subscribe(
```

Comment:
the second parameter of subscribe is a catch.
It seems that the [catch operator](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/catch.md) is only used to catch errors until a good result come.